beesdoo_shift_swap/models/dated_template.py

```diff
@@ -0,0 +1,175 @@
+
+        shift_generated = (self.env["beesdoo.shift.shift"].sudo().search([
+            ("start_time", ">", start_date.strftime("%Y-%m-%d %H:%M:%S"))
+        ],
+        order="start_time, task_template_id, task_type_id"))
+
+        timeslot_rec = self.swap_shift_to_timeslot(shift_generated)
+
+        # generate timeslot of the shift not generated
+        last_sequence = int(
+            self.env["ir.config_parameter"]
+                .sudo()
+                .get_param("last_planning_seq")
+        )
+        next_planning = self.env["beesdoo.shift.planning"]._get_next_planning(last_sequence)
+        next_planning_date = fields.Datetime.from_string(
+            self.env["ir.config_parameter"]
+                .sudo()
+                .get_param("next_planning_date", 0)
+        )
+        #TODO : create system parameters for end_date
+        next_swap_limit = int(
+            self.env["ir.config_parameter"]
+                .sudo()
+                .get_param("beesdoo_shift.day_limit_swap")
+        )
+        end_date = my_timeslot.date + timedelta(days=next_swap_limit)
+        next_planning = next_planning.with_context(visualize_date=next_planning_date)
+        shift_recset = self.env["beesdoo.shift.shift"]
+
+        '''
+        if not next_planning.task_template_ids:
+            _logger.error(
+                "Could not generate next planning: no task template defined."
+            )
+            return
+        '''
+
+        while next_planning_date < end_date :
+            shift_recset = next_planning.task_template_ids._generate_task_day()
+            timeslot_rec |= self.swap_shift_to_timeslot(shift_recset)
+            next_planning_date = next_planning._get_next_planning_date(next_planning_date)
+            last_sequence = next_planning.sequence
+            next_planning = self.env["beesdoo.shift.planning"]._get_next_planning(last_sequence)
+            next_planning = next_planning.with_context(visualize_date=next_planning_date)
+
+        return timeslot_rec
+
+
+    #TODO: show my next timeslot/use myshift_next_shift + swap_shift_to_timeslot
+    @api.multi
+    def my_timeslot(self, worker_id):
+        shifts = worker_id.my_next_shift()
+        timeslots = self.swap_shift_to_timeslot(shifts)
+        return timeslots
+
+class TaskTemplate(models.Model):
+
+    _inherit = "beesdoo.shift.template"
+
+    @api.multi
+    def _generate_task_day(self):
+        shifts = super(TaskTemplate,self)._generate_task_day()
+        exchanges = self.env["beesdoo.shift.subscribed_underpopulated_shift"].search([])
+        template={"first":None,"second": None}
+        for shift in shifts :
+            template["first"] = shift.task_template_id
+            for exchange in exchanges :
+                if shift.worker_id.name == False and exchange.confirmed_timeslot_id.template_id == shift.task_template_id and shift.start_time == exchange.confirmed_timeslot_id.date and not exchange.confirme_status:
```

Comment:
I think that you can use `not shift.worker_id and ...` instead of `shift.worker_id.name == False`.